fairseq/tasks/hubert_pretraining.py

```diff
@@ -138,14 +138,22 @@ def dictionaries(self) -> List[Dictionary]:
     def setup_task(
         cls, cfg: HubertPretrainingConfig, **kwargs
     ) -> "HubertPretrainingTask":
+        return cls(cfg)
+        
+    def dictionaries_factory(self, cfg: HubertPretrainingConfig):
         label_dir = cfg.data if cfg.label_dir is None else cfg.label_dir
-        dictionaries = {
+        return {
             label: Dictionary.load(f"{label_dir}/dict.{label}.txt")
             if os.path.exists(f"{label_dir}/dict.{label}.txt")
             else None
             for label in cfg.labels
         }
-        return cls(cfg, dictionaries)
+
+    def target_dictionary_factory(self):
+        if len(self._dictionaries) == 1:
+            logger.info(self._dictionaries)
+            return self._dictionaries[list(self._dictionaries)[0]]
+        return None
```

Comment:
May be returning {} is better to not crash the process, and signal that the target dictionary is empty.